Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -1673,30 +1619,26 @@ JS_DEFINE_NATIVE_FUNCTION(TypedArrayPrototype::to_sorted)
     // 3. Perform ? ValidateTypedArray(O).
     auto* typed_array = TRY(validate_typed_array_from_this(vm));
 
-    // 4. Let buffer be obj.[[ViewedArrayBuffer]].
-    auto* array_buffer = typed_array->viewed_array_buffer();
-    VERIFY(array_buffer);
-
-    // 5. Let len be O.[[ArrayLength]].
+    // 4. Let len be O.[[ArrayLength]].
     auto length = typed_array->array_length();
 
-    // 6. Let A be ? TypedArrayCreateSameType(O, « 𝔽(len) »).
+    // 5. Let A be ? TypedArrayCreateSameType(O, « 𝔽(len) »).
     MarkedVector<Value> arguments(vm.heap());
     arguments.empend(length);
     auto* return_array = TRY(typed_array_create_same_type(vm, *typed_array, move(arguments)));
 
-    // 7. NOTE: The following closure performs a numeric comparison rather than the string comparison used in  Array.prototype.toSorted
-    // 8. Let SortCompare be a new Abstract Closure with parameters (x, y) that captures comparefn and buffer and performs the following steps when called:
+    // 6. NOTE: The following closure performs a numeric comparison rather than the string comparison used in  Array.prototype.toSorted
+    // 7. Let SortCompare be a new Abstract Closure with parameters (x, y) that captures comparefn and performs the following steps when called:
     Function<ThrowCompletionOr<double>(Value, Value)> sort_compare = [&](auto x, auto y) -> ThrowCompletionOr<double> {
-        // a. Return ? CompareTypedArrayElements(x, y, comparefn, buffer).
-        return TRY(compare_typed_array_elements(vm, x, y, comparefn.is_undefined() ? nullptr : &comparefn.as_function(), *return_array->viewed_array_buffer()));
```

Comment:
:eyes: there was a bug here, `return_array->viewed_array_buffer()` was the wrong buffer (drove me nuts trying to figure out why the test added below wasn't throwing)